### ISSUE TITLE
Failed to install matplotlib==3.6.0 (or other versions tried)

### ISSUE THREAD

**Issue by jamescorbe** [chaquo/chaquopy#841]
**Issue**
Cant install matplotlib versions.
The error states that cython is not found however it is installed and included in my requirements file,  also not able to install numpy newer than the 1.19.5 which is installed when no version is specified.

if no matplotlib , or numpy version is specified I have version issues from other dependencies. I have tried installing directly from the correct .whl files from https://chaquo.com/pypi-7.0/numpy/  and https://chaquo.com/pypi-7.0/matplotlib/ but it has the same issue.

```
ultralytics 8.0.58 has requirement numpy>=1.21.6, but you'll have numpy 1.19.5 which is incompatible.
ultralytics 8.0.58 has requirement matplotlib>=3.2.2, but you'll have matplotlib 3.1.2 which is incompatible.
```

**requirements**
gTTS==2.3.1
Cython==0.29.30
playsound==1.2.2
pysine==0.9.2
numpy==1.23.3
matplotlib==3.6.0
supervision==0.3.2
opencv-python==4.7.0.72
ultralytics==8.0.58

**Chaquopy version**
id 'com.chaquo.python' version '14.0.2' apply false

**Relevant parts of your build.gradle file**

  ndk {
            abiFilters "armeabi-v7a", "arm64-v8a", "x86", "x86_64"
        }

python {
            version "3.8"
            buildPython "C:/Users/james/AppData/Local/Programs/Python/Python38/python.exe"
            pip{
                install "torchvision"
                install "-r", "C:\\Users\\james\\AndroidStudioProjects\\ODVI\\app\\src\\main\\python\\requirements.txt"
            }
        }

**Describe your issue**
```
Collecting matplotlib==3.6.0
  Using cached matplotlib-3.6.0.tar.gz (35.7 MB)
    ERROR: Command errored out with exit status 1:
     command: 'C:\Users\james\AppData\Local\Programs\Python\Python38\python.exe' -S -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\james\\AppData\\Local\\Temp\\pip-install-nlunkdz6\\matplotlib\\setup.py'"'"'; __file__='"'"'C:\\Users\\james\\AppData\\Local\\Temp\\pip-install-nlunkdz6\\matplotlib\\setup.py'"'"';import chaquopy_monkey; chaquopy_monkey.disable_native();f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' --no-user-cfg egg_info --egg-base 'C:\Users\james\AppData\Local\Temp\pip-pip-egg-info-vqm_w695'
         cwd: C:\Users\james\AppData\Local\Temp\pip-install-nlunkdz6\matplotlib\
    Complete output (101 lines):
    Running from numpy source directory.
    C:\Users\james\AppData\Local\Temp\easy_install-sp5lnbct\numpy-1.24.2\setup.py:67: DeprecationWarning:
    ERROR: Command errored out with exit status 1:

    
      `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
      of the deprecation of `distutils` itself. It will be removed for
      Python >= 3.12. For older Python versions it will remain present.
      It is recommended to use `setuptools < 60.0` for those Python versions.
      For more details, see:
        https://numpy.org/devdocs/reference/distutils_status_migration.html
    
    
      import numpy.distutils.command.sdist
    Traceback (most recent call last):
      File "C:\Users\james\AppData\Local\Temp\easy_install-sp5lnbct\numpy-1.24.2\setup.py", line 233, in generate_cython
        "suppress output with --quiet.")
    ModuleNotFoundError: No module named 'Cython'
    
    The above exception was the direct cause of the following exception:
    
    Traceback (most recent call last):
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 154, in save_modules
        yield saved
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 195, in setup_context
        yield
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 250, in run_setup
        _execfile(setup_script, ns)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 45, in _execfile
        exec(code, globals, locals)
      File "C:\Users\james\AppData\Local\Temp\easy_install-sp5lnbct\numpy-1.24.2\setup.py", line 479, in <module>
      File "C:\Users\james\AppData\Local\Temp\easy_install-sp5lnbct\numpy-1.24.2\setup.py", line 458, in setup_package
      File "C:\Users\james\AppData\Local\Temp\easy_install-sp5lnbct\numpy-1.24.2\setup.py", line 240, in generate_cython
        message = package.check()
    OSError: Cython needs to be installed in Python as a module
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\james\AppData\Local\Temp\pip-install-nlunkdz6\matplotlib\setup.py", line 260, in <module>
        setup(  # Finally, pass this all along to setuptools to do the heavy lifting.
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\__init__.py", line 144, in setup
        _install_setup_requires(attrs)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\__init__.py", line 139, in _install_setup_requires
        dist.fetch_build_eggs(dist.setup_requires)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\dist.py", line 717, in fetch_build_eggs
        resolved_dists = pkg_resources.working_set.resolve(
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\pkg_resources\__init__.py", line 782, in resolve
        dist = best[req.key] = env.best_match(
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\pkg_resources\__init__.py", line 1067, in best_match
        return self.obtain(req, installer)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\pkg_resources\__init__.py", line 1079, in obtain
        return installer(requirement)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\dist.py", line 787, in fetch_build_egg
        return cmd.easy_install(req)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\command\easy_install.py", line 679, in easy_install
        return self.install_item(spec, dist.location, tmpdir, deps)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\command\easy_install.py", line 705, in install_item
        dists = self.install_eggs(spec, download, tmpdir)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\command\easy_install.py", line 890, in install_eggs
        return self.build_and_install(setup_script, setup_base)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\command\easy_install.py", line 1158, in build_and_install
        self.run_setup(setup_script, setup_base, args)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\command\easy_install.py", line 1144, in run_setup
        run_setup(setup_script, args)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 253, in run_setup
        raise
      File "C:\Users\james\AppData\Local\Programs\Python\Python38\lib\contextlib.py", line 131, in __exit__
        self.gen.throw(type, value, traceback)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 195, in setup_context
        yield
      File "C:\Users\james\AppData\Local\Programs\Python\Python38\lib\contextlib.py", line 131, in __exit__
        self.gen.throw(type, value, traceback)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 166, in save_modules
        saved_exc.resume()
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 141, in resume
        six.reraise(type, exc, self._tb)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\_vendor\six.py", line 685, in reraise
        raise value.with_traceback(tb)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 154, in save_modules
        yield saved
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 195, in setup_context
        yield
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 250, in run_setup
        _execfile(setup_script, ns)
      File "C:\Users\james\AndroidStudioProjects\ODVI\app\build\generated\python/bp\setuptools\sandbox.py", line 45, in _execfile
        exec(code, globals, locals)
      File "C:\Users\james\AppData\Local\Temp\easy_install-sp5lnbct\numpy-1.24.2\setup.py", line 479, in <module>
      File "C:\Users\james\AppData\Local\Temp\easy_install-sp5lnbct\numpy-1.24.2\setup.py", line 458, in setup_package
      File "C:\Users\james\AppData\Local\Temp\easy_install-sp5lnbct\numpy-1.24.2\setup.py", line 240, in generate_cython
        message = package.check()
    OSError: Cython needs to be installed in Python as a module
    
    Edit mplsetup.cfg to change the build options; suppress output with --quiet.
    
    BUILDING MATPLOTLIB
          python: yes [3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC
                      v.1928 64 bit (AMD64)]]
        platform: yes [win32]
           tests: no  [skipping due to configuration]
          macosx: no  [Mac OS-X only]
    
    ----------------------------------------
ERROR: Failed to install matplotlib==3.6.0 from https://files.pythonhosted.org/packages/69/e6/c36374904b757c8193a44af789ddf5bc27f2fe5fbd0cdd908f09cb21e2e1/matplotlib-3.6.0.tar.gz#sha256=c5108ebe67da60a9204497d8d403316228deb52b550388190c53a57394d41531 (from -r C:\Users\james\AndroidStudioProjects\ODVI\app\src\main\python\requirements.txt (line 6)).
For assistance, please raise an issue at https://github.com/chaquo/chaquopy/issues.
Failed to install matplotlib==3.6.0 from https://files.pythonhosted.org/packages/69/e6/c36374904b757c8193a44af789ddf5bc27f2fe5fbd0cdd908f09cb21e2e1/matplotlib-3.6.0.tar.gz#sha256=c5108ebe67da60a9204497d8d403316228deb52b550388190c53a57394d41531 (from -r C:\Users\james\AndroidStudioProjects\ODVI\app\src\main\python\requirements.txt (line 6)).

Chaquopy: Exit status 1

> Task :app:generateDebugPythonRequirements FAILED
Execution failed for task ':app:generateDebugPythonRequirements'.
> Process 'command 'C:/Users/james/AppData/Local/Programs/Python/Python38/python.exe'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

```

**Comment by mhsmith** [comment]
We've just released matplotlib 3.6.0 for Python 3.8. Previously this version was only available for Python 3.9 and 3.10.

There are a couple of other obstacles to installing ultralytics:
* ultralytics 8.0.58 requires `opencv-python>=4.6.0`, but our current version is 4.5.1. This can be resolved by going back to ultralytics version 8.0.16 or older.
* ultralytics requires `tensorboard>=2.4.1`, whose current version requires `grpcio>=1.48.2`, but our current version is 1.39.0. This can be resolved by pinning a specific version of tensorboard (see [#843](https://github.com/chaquo/chaquopy/issues/843)).

So to install ultralytics, update your requirements as follows:
```
                install "ultralytics==8.0.16"
                install "tensorboard==2.11.2"
```
And remove your requirements for numpy, matplotlib, and opencv-python.

**Comment by mhsmith** [comment]
There are also a couple of issues with your other requirements:

`pysine` requires `pyaudio`, which we don't currently support (#389).

`supervision` requires `numpy>=1.20.0`, which for Python 3.8 requires adding the following line to your `pip` block (https://github.com/chaquo/chaquopy/issues/559#issuecomment-989318791):
```
options "--extra-index-url", "https://chaquo.com/pypi-extra/"
```
